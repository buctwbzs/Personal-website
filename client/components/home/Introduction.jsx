import React, { Component } from 'react'
import styled from 'styled-components'

const IntroductionBox = styled.section`
  display: flex;
  width: 100%;
  height: 100px;
  >p{
    margin: 0;
    padding: 1em;
    line-height: 1.5em;
    font-size: 1.2rem;
    font-style: italic;
    text-align: justify;
    word-break: break-all;
    word-wrap: break-word;
    >a {
      color: #AD6800;
      cursor: pointer;
      transition: color .2s ease-in;
      &:hover {
        color: #613400;
      }
    }
  }
  @media (max-width: 899px) {
      height: 140px;
  }
  @media (max-width: 499px) {
      height: 180px;
  }
`
class Introduction extends Component {
  render() {
    return (
      <IntroductionBox>
        <p>
          This is my personal website. There are three versions of this website:&nbsp;
          <a href="#">React</a>,&nbsp;
          <a href="http://angular.buctwbzs.com">Angular</a>,&nbsp;
          <a href="#">Vue</a>;&nbsp;
          all websites use server-side rendering, and the source code is open(github).
          The current site built with <b>React</b>.
          If necessary,i hope you can create some new issues and help me improve them.
        </p>
      </IntroductionBox>
    )
  }
}



export default Introduction