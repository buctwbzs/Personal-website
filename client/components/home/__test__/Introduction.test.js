import React from 'react'
import renderer from 'react-test-renderer'
import Introduction from '../Introduction'

it('Introductoin renders correctly', () => {
  const tree = renderer.create(<Introduction />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})