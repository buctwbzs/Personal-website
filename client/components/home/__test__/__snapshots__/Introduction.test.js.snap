// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Introductoin renders correctly 1`] = `
<section
  className="Introduction__IntroductionBox-s1hu0lqs-0 iJrNfr"
>
  <p>
    This is my personal website. There are three versions of this website: 
    <a
      href="#"
    >
      React
    </a>
    , 
    <a
      href="http://angular.buctwbzs.com"
    >
      Angular
    </a>
    , 
    <a
      href="#"
    >
      Vue
    </a>
    ;  all websites use server-side rendering, and the source code is open(github). The current site built with React. If necessary,i hope you can create some new issues and help me improve them.
  </p>
</section>
`;
